java/org/apache/catalina/realm/JNDIRealm.java

```diff
@@ -469,6 +474,19 @@
      */
     protected boolean useContextClassLoader = true;
 
+    /**
+     * The comma separated names of user attributes to additionally query from the
+     * user's directory entry. These will be provided to the user through the
+     * created Principal's <i>attributes</i> map.
+     */
+    protected String userAttributes;
```

Comment:
Same as in the `DataSourceRealm`

Reply:
Same solution as in the `DataSourceRealm`?

Reply:
Agreed.